chia/data_layer/data_layer_wallet.py

```diff
@@ -102,7 +103,7 @@ def from_json_dict(cls, json_dict: Dict[str, Any]) -> "Mirror":
 
 
 @final
-class DataLayerWallet:
+class DataLayerWallet(WalletProtocol):
```

Comment:
to repeat what kyle said. This is not how you declare that you *implement* a protocol, this is how you declare a *new* protocol that subsumes another.

My understanding is that whether a type implements a protocol is essentially duck-typing in mypy

Reply:
also known as structural subtyping, in case you see that floating around.  i still intend to get a smaller pattern working someday but at the moment mypy doesn't provide verbose output clarifying in what way the protocol is not satisfied when you do it my 'better' way.